de-lib/src/main/java/org/iplantc/de/preferences/client/view/PreferencesViewImpl.java

```diff
@@ -166,25 +224,60 @@ public boolean isValid() {
                             ks.markInvalid(appearance.duplicateShortCutKey(kbMap.get(ks)));
                             sc.markInvalid(appearance.duplicateShortCutKey(kbMap.get(ks)));
                             valid = false;
+                            IplantAnnouncer.getInstance()
+                                           .schedule(new ErrorAnnouncementConfig(appearance.completeRequiredFieldsError()));
                         }
                     }
                 }
             }
         }
+        if (!Strings.isNullOrEmpty(hookUrl.getValue())) {
+            List<String> topics = getSelectedTopics();
+            if (topics == null || topics.size() == 0) {
+                IplantAnnouncer.getInstance()
+                               .schedule(new ErrorAnnouncementConfig(appearance.mustSelectATopic()));
+                valid = false;
+            }
+        }
+
+        if (editorDriver.hasErrors()) {
+            IplantAnnouncer.getInstance()
```

Comment:
Same with this `getInstance()`, we should switch to an injected `IplantAnnouncer`